Gems/O3DE/GeomNodes/Code/Source/Editor/Systems/GeomNodesSystem.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "GeomNodesSystem.h"
+#include "Editor/UI/PropertyFileSelect.h"
+#include "Editor/UI/ValidationHandler.h"
+#include "Editor/UI/GeomNodesValidator.h"
+#include <Editor/Common/GNAPI.h>
+#include <Editor/EBus/IpcHandlerBus.h>
+
+namespace GeomNodes
+{
+    long IpcHandlerCB(AZ::u64 id, const char* data, AZ::u64 length)
+    {
+        /*AZStd::string msg;
+        msg.assign(data, length);
+        AZ_Printf("GeomNodesSystem", "id: %llu data: %s length: %llu", id, msg.c_str(), length);*/
+        Ipc::IpcHandlerNotificationBus::Event(
+            AZ::EntityId(id), &Ipc::IpcHandlerNotificationBus::Events::OnMessageReceived, reinterpret_cast<const AZ::u8*>(data), length);
+
+        return 0;
+    }
+
+    GeomNodesSystem::GeomNodesSystem(AZStd::unique_ptr<GNSettingsRegistryManager> registryManager)
+        : m_registryManager(AZStd::move(registryManager))
+        , m_propertyHandlers()
+        , m_validator(AZStd::make_unique<Validator>())
+        , m_validationHandler(AZStd::make_unique<ValidationHandler>())
+        {
+        }
+
+    GeomNodesSystem::~GeomNodesSystem()
+    {
+        Shutdown();
+    }
+
+    void GeomNodesSystem::Initialize(const GNConfiguration* config)
+    {
+		if (m_state == State::Initialized)
+		{
```

Comment:
inconsistent number of indentation spaces throughout file

Reply:
fixed on commit [d756b38](https://github.com/o3de/o3de-technicalart/pull/11/commits/d756b38541172fd3e0051140ca96cc9bd8b05388)